docs/examples/oceanmodelling.py

```diff
@@ -0,0 +1,509 @@
+#
+
+
+# %%
+def initialise_gp(kernel, mean, dataset):
+    prior = gpx.Prior(mean_function=mean, kernel=kernel)
+    likelihood = gpx.Gaussian(
+        num_datapoints=dataset.n, obs_noise=jnp.array([1.0e-6], dtype=jnp.float64)
+    )
+    posterior = prior * likelihood
+    return posterior
+
+
+# Define the velocity GP
+mean = gpx.mean_functions.Zero()
+kernel = VelocityKernel()
+velocity_posterior = initialise_gp(kernel, mean, dataset_train)
+
+
+# %% [markdown]
+# With a model now defined, we can proceed to optimise the hyperparameters of our likelihood over $D_0$. This is done by minimising the marginal log likelihood using `optax`. We also plot its value at each step to visually confirm that we have found the minimum. See the  [introduction to Gaussian Processes](https://docs.jaxgaussianprocesses.com/examples/intro_to_gps/) notebook for more information on optimising the MLL.
+
+
+# %%
+def optimise_mll(posterior, dataset, NIters=1000, key=key, plot_history=True):
+    # define the Marginal Log likelihood using dataset_train
+    objective = gpx.objectives.ConjugateMLL(negative=True)
+    # Optimise to minimise the MLL
+    optimiser = ox.adam(learning_rate=0.1)
+    opt_posterior, history = gpx.fit(
+        model=posterior,
+        objective=objective,
+        train_data=dataset,
+        optim=optimiser,
+        num_iters=NIters,
+        safe=True,
+        key=key,
+    )
+    # plot MLL value at each iteration
+    if plot_history:
+        fig, ax = plt.subplots(1, 1)
+        ax.plot(history, color=colors[1])
+        ax.set(xlabel="Training iteration", ylabel="Negative marginal log likelihood")
+
+    return opt_posterior
+
+
+opt_velocity_posterior = optimise_mll(velocity_posterior, dataset_train)
+
+
+# %% [markdown]
+# ### Comparison
+# We next obtain the latent distribution of the GP of $g$ at $\mathbf{x}_{0,i}$, then extract its mean and standard at the test locations, $\mathbf{F}_{\text{latent}}(\mathbf{x}_{0,i})$, as well as the standard deviation (we will use it at the very end).
+
+
+# %%
+def latent_distribution(opt_posterior, pos_3d, dataset_train):
+    latent = opt_posterior.predict(pos_3d, train_data=dataset_train)
+    latent_mean = latent.mean()
+    latent_std = latent.stddev()
+    return latent_mean, latent_std
+
+
+# extract latent mean and std of g, redistribute into vectors to model F
+velocity_mean, velocity_std = latent_distribution(
+    opt_velocity_posterior, dataset_ground_truth.X, dataset_train
+)
+
+dataset_latent_velocity = dataset_3d(pos_test, velocity_mean)
+
+
+# %% [markdown]
+# We now replot the ground truth (testing data) $D_0$, the predicted latent vector field $\mathbf{F}_{\text{latent}}(\mathbf{x_i})$, and a heatmap of the residuals at each location $\mathbf{R}(\mathbf{x}_i) = \mathbf{y}_{0,i} - \mathbf{F}_{\text{latent}}(\mathbf{x}_i) $, as well as $\left|\left|\mathbf{R}(\mathbf{x}_i)\right|\right|$.
```

Comment:
What is `\mathbf{x_i}` in `$\mathbf{F}_{\text{latent}}(\mathbf{x_i})$` and `$\mathbf{R}(\mathbf{x}_i) = \mathbf{y}_{0,i} - \mathbf{F}_{\text{latent}}(\mathbf{x}_i) $` and $\left|\left|\mathbf{R}(\mathbf{x}_i)\right|\right|$ - this does not seemed to be defined? Is this \mathbf{x}_{0,i}?